cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -247,26 +251,28 @@ public static void sanityCheckCapacityEstimation(boolean allowCapacityEstimation
 
   /**
    * Rebalance the cluster
-   * @param goals the goals to be met during the rebalance. When empty all goals will be used.
-   * @param dryRun whether it is a dry run or not.
+   * @param goals The goals to be met during the rebalance. When empty all goals will be used.
+   * @param dryRun Whether it is a dry run or not.
    * @param requirements The cluster model completeness requirements.
-   * @param operationProgress the progress of the job to report.
+   * @param operationProgress The progress of the job to report.
    * @param allowCapacityEstimation Allow capacity estimation in cluster model if the requested broker capacity is unavailable.
    * @param concurrentPartitionMovements The maximum number of concurrent partition movements per broker
    *                                     (if null, use num.concurrent.partition.movements.per.broker).
    * @param concurrentLeaderMovements The maximum number of concurrent leader movements
    *                                  (if null, use num.concurrent.leader.movements).
-   * @return the optimization result.
-   * @throws KafkaCruiseControlException when the rebalance encounter errors.
+   * @param skipHardGoalCheck Whether check and add all hard goals to {@code goals}.
```

Comment:
If we rephrase it as follows, would it be more clear?
`Whether check and add all hard goals to {@code goals}.` -> `True if the provided {@code goals} do not have to contain all hard goals, false otherwise`